src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginServiceHelpers.cs

```diff
@@ -194,89 +220,168 @@ internal static class InheritanceMarginServiceHelper
+                    solution,
+                    symbol,
+                    InheritanceRelationship.ImplementingMember,
+                    cancellationToken), cancellationToken).ConfigureAwait(false);
 
-            // Right now the targets are not shown in a classified way.
-            var definition = await targetSymbol.ToNonClassifiedDefinitionItemAsync(
-                solution,
-                includeHiddenLocations: false,
-                cancellationToken: cancellationToken).ConfigureAwait(false);
+            return new SerializableInheritanceMarginItem(
+                lineNumber,
+                FindUsagesHelpers.GetDisplayParts(memberSymbol),
+                memberSymbol.GetGlyph(),
+                implementedMemberItems);
+        }
 
-            var displayName = targetSymbol.ToDisplayString(s_displayFormat);
+        private static async ValueTask<SerializableInheritanceMarginItem> CreateInheritanceItemForClassAndStructureAsync(
+            Solution solution,
+            INamedTypeSymbol memberSymbol,
+            int lineNumber,
+            ImmutableArray<ISymbol> baseSymbols,
+            ImmutableArray<ISymbol> derivedTypesSymbols,
+            CancellationToken cancellationToken)
+        {
+            // If the target is an interface, it would be shown as 'Inherited interface',
+            // and if it is an class/struct, it whould be shown as 'Base Type'
+            var baseSymbolItems = await baseSymbols
+                .SelectAsArray(symbol => symbol.OriginalDefinition)
+                .Distinct()
+                .SelectAsArrayAsync((symbol, _) =>
+                    symbol.IsInterfaceType()
+                        ? CreateInheritanceItemAsync(
+                            solution,
+                            symbol,
+                            InheritanceRelationship.ImplementedInterface, cancellationToken)
+                        : CreateInheritanceItemAsync(
+                            solution,
+                            symbol,
+                            InheritanceRelationship.BaseType, cancellationToken), cancellationToken)
+                .ConfigureAwait(false);
 
-            return new SerializableInheritanceTargetItem(
-                inheritanceRelationship,
-                // Id is used by FAR service for caching, it is not used in inheritance margin
-                SerializableDefinitionItem.Dehydrate(id: 0, definition),
-                targetSymbol.GetGlyph(),
-                displayName);
+            var derivedTypeItems = await derivedTypesSymbols
+                .SelectAsArray(symbol => symbol.OriginalDefinition)
+                .Distinct()
+                .SelectAsArrayAsync((symbol, _) =>
+                    CreateInheritanceItemAsync(solution,
+                    symbol,
+                    InheritanceRelationship.DerivedType,
+                    cancellationToken), cancellationToken)
```

Comment:
please indent children *in* from parents.  otherwise, they look like siblings :)

Reply:
I think I might be confused by the inline parameter 
![image](https://user-images.githubusercontent.com/24360909/122097387-7b3fe080-cdc4-11eb-97a9-2a36459c362e.png)

The text makes me feel the parameters are already indented somehow. 🙁

FYI @akhera99 , this might not meet the bar of a bug, but just let you aware this problem.

Reply:
Interesting. Visually I still see it as not indented because the text starts on the same vertical space. I can see how you got confused though